app/models/defence_request_transitions/acknowledge.rb

```diff
@@ -18,7 +18,7 @@ def set_cco
     end
 
     def acknowledge_defence_request
-      run_transition && defence_request.save!
+      run_transition && defence_request.save! && defence_request.generate_dscc_number!
```

Comment:
Good suggestion, I think we want to disuade people from calling `defence_request.generate_dscc_number!` without save as much as possible, this would allow us to make it private.